### PR TITLE
Added shell script for server starting and stopping

### PR DESCRIPTION
**Pull request by hartmannaf**
(normaly uses start stop deamon, but also has alternativ using lsof)
this allows us to use pid files, without the need to manage them by our own.

### CODE REVIEW

File: drop-server.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+pythonpath=$(command -v python2 || command -v python)
+startstopdeamon="/sbin/start-stop-daemon"
```
Comment (thechauffeur) [inline drop-server.sh]: Not everyone uses a debian based system.

Comment (thechauffeur) [inline drop-server.sh]: Ok at least `start-stop-daemon` is not required. Which does not make the `lsof` hack better but anyway.

**Comment by thechauffeur** [comment]
@hartmannaf I want to leave some words about this approach.

The patch looks like you want to provide an init script. For example this looks like it because this `start-stop-daemon` resided in `/sbin`. Many problems come with this. `start-stop-daemon` may not be provided by the distribution a user works on. There are unix/linux system out there which are not designed to have normal users run executables from `/sbin` or `/usr/sbin`. Our drop server can easily be run without any privileges except for opening sockets with a port number high enough to not cause any problems.

An init script heavily depends on the distribution used or at least on the init system used in that distribution. This is exactly the reason why scripts like these are often not provided by upstream (i.e. us regarding `qabel-drop`) but instead by the packager maintainer of each distribution. That said, for init systems like systemd it is often enough if upstream provides a unit file because it works on almost all distributions using systemd.

In any case providing an init script is not really what we need at the moment. When I want to run the drop server I would just start it in my virtualenv. If I want to do it it more professionally, I would use gunicorn (virtualenv or not) using my users privileges.

If you provide this patch for the sole reason you want to use it in the gradle scripts in the `qabel` repository, I think this approach is the wrong place (e.g. this would not work on my machine so it would fall back to the `lsof` hack).

Don't get me wrong. You can provide init scripts for as much operating systems as you want but the name and description of this PR reads like you want to provide an general approach -- which you don't do with this patch.

File: drop-server.sh (hunk 1 above)
Comment (thechauffeur) [inline drop-server.sh]: See https://github.com/Qabel/qabel-drop/pull/19#issuecomment-114172949

**Comment by hartmannaf** [comment]
In general this should work on all unix/linux system, but ist still uses the lsof workaround at the moment. 
I could try to replace the lsof fallback by netstat, than it should work an nearly all unix/linux systems.
I could also add a geunicorn fallback before using the last port fallback. 
Also, why not serve it generely with the servers, when its needed for the gradlescript, it only uses normal sh functions and checks if the used program is available, so other people could maybe use it too.

**Comment by thechauffeur** [comment]
> In general this should work on all unix/linux system, but ist still uses the lsof workaround at the moment.
Like I said.

> I could try to replace the lsof fallback by netstat, than it should work an nearly all unix/linux systems.

replacing `lsof` with `netstat` would not help.

> I could also add a geunicorn fallback before using the last port fallback.

Maybe. You could also use `gunicorn` in the gradle scripts and add it to the requirements.

> Also, why not serve it generely with the servers, when its needed for the gradlescript, it only uses normal sh functions and checks if the used program is available, so other people could maybe use it too.
I didn't say you should not provide this. Of course you can do it -- go ahead. I just mentioned that you are not addressing the gradle testing problem with this PR (on all linux machines) and since you didn't reference any issues which are solved by or helped with this PR I don't know your intention.

**Pull request merged**